lib/components/component.dart

```diff
@@ -34,38 +34,41 @@ abstract class Component {
   /// Use [Resizable] to save the gameSize in a component.
   void onGameResize(Vector2 gameSize) {}
 
+  /// Whether this component should be removed or not.
+  ///
+  /// It will be checked once per component per tick, and if it is true, [BaseGame] will remove it.
+  bool shouldRemove = false;
+
+  /// Remove the component from the game it is added to in the next tick
+  void remove() => shouldRemove = true;
```

Comment:
Should we make `shouldRemove` private since we have a method for marking it for removal?

Reply:
I used `shouldRemove` just as a variable with the intention to replace the old `destroy()` method, so it's that value that `BaseGame` looks at when it is removing components from the component list.